test/functional/run/global_destructor.glu

```diff
@@ -0,0 +1,29 @@
+// RUN: gluc %s -o %t && %t | FileCheck %s
```

Comment:
Add a test where the global is never initialized (shouldn't be destroyed), and a test for the above edge case